src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,208 @@
+"""GitHub App Credential Plugin.
+
+This module defines a credential plugin for GitHub Apps, allowing
+authentication via GitHub App tokens.
+
+Functions:
+- github_app_backend: Generates a GitHub App token fields from GitHub Admin.
+- github_app_lookup: Defines the credential plugin interface.
+"""
+# FIXME: ManagedCredentialType type not found.
+
+from typing import TypedDict
+
+from awx_plugins.interfaces._temporary_private_api import (  # noqa: WPS436
+    ManagedCredentialType,
+)
+from awx_plugins.interfaces._temporary_private_django_api import (  # noqa: WPS436
+    gettext_noop as _,
+)
+
+from github import Auth, Github
+
+from .injectors import github_app_injector
+from .plugin import CredentialPlugin
+
+
+# Define expected structure for 'fields'
+class FieldDict(TypedDict, total=False):
+    id: str
+    label: str
+    type: str
+    format: str  # Optional field
+    secret: bool  # Optional field
+    multiline: bool  # Optional field
```

Comment:
did you mean this by "optional"?
```suggestion
    format: str | None
    secret: bool | None
    multiline: bool | None
```